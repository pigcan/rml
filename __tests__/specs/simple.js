const MLTransformer = require('../../src/Transformer');

describe('MLTransformer', () => {
  it('style/className/class can not be bool', (done) => {
    new MLTransformer(
      `
<a style/>
<a class/>
<a className/>
`.trim()
    ).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });

  it('escape attribute', (done) => {
    new MLTransformer(
      `
<a x='"\`'>"'\`</a>
`.trim()
    ).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });

  it('support simple', (done) => {
    new MLTransformer([
      '<div prop="{{a+b}}" x="1" onClick="{{this.onClick}}">',
      '<div>',
      '1{{a[0].b}}2',
      '</div>',
      '<div>',
      '1',
      '</div>',
      '<div>',
      '1',
      '</div>',
      '</div>',
    ].join('\n')).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });

  it('support ignore attr value', (done) => {
    new MLTransformer([
      `<div checked>`,
      `</div>`,
    ].join('\n')).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });

  it('support bool attr value', (done) => {
    new MLTransformer([
      `<div checked="{{false}}">`,
      `</div>`,
    ].join('\n')).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });

  it('allowImportModule', (done) => {
    new MLTransformer([
      `<import-module name="X" from="y" />`,
      `<import-module name="{Z, Q:Y}" from="y" />`,
      `<X value="{{Y}}"><Y/><Z/></X>`,
    ].join('\n'), {
      allowImportModule: true,
    }).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });

  it('allowScript', (done) => {
    new MLTransformer([
      `<script>var x = 1;</script>`,
      `<X/>`,
    ].join('\n'), {
      allowScript: true,
    }).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });

  it('skip comment', (done) => {
    new MLTransformer(
      `
<!-- 1 -->
<a />
<!-- 2 -->
`.trim()
    ).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });
});
