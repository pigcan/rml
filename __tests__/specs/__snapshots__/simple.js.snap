// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MLTransformer allowImportModule 1`] = `
"import React from 'react';
import X from 'y';
import { Z, Q as Y } from 'y';

export default function render(data) {
  return (
    <X
      value = {(Y)}
    >
      <Y>
      </Y>
      <Z>
      </Z>
    </X>
  );
};"
`;

exports[`MLTransformer allowScript 1`] = `
"import React from 'react';
var x = 1;

export default function render(data) {
  return (
    <X>
    </X>
  );
};"
`;

exports[`MLTransformer do not change number 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <a
      x = {\`123\`}
    >
    </a>
  );
};"
`;

exports[`MLTransformer escape attribute 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <a
      x = {\`\\"\\\\\`\\\\$\`}
    >
      {\`\\"'\\\\\`\\\\$\`}
    </a>
  );
};"
`;

exports[`MLTransformer explicit number 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <a
      x = {(123)}
    >
    </a>
  );
};"
`;

exports[`MLTransformer skip comment 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <a>
    </a>
  );
};"
`;

exports[`MLTransformer style/className/class can not be bool 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    [
    <a>
    </a>
    ,
    <a>
    </a>
    ,
    <a>
    </a>
    ,
    ]
  );
};"
`;

exports[`MLTransformer support bool attr value 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <div
      checked = {(false)}
    >
    </div>
  );
};"
`;

exports[`MLTransformer support ignore attr value 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <div
      checked
    >
    </div>
  );
};"
`;

exports[`MLTransformer support simple 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <div
      prop = {(data.a + data.b)}
      x = {\`1\`}
      onClick = {(this.onClick)}
    >
      <div>
        {[\`1\`, (data.a[0].b), \`2\`]}
      </div>
      <div>
        {\`1\`}
      </div>
      <div>
        {\`1\`}
      </div>
    </div>
  );
};"
`;
